lib/Amin/Command/Du.pm

```diff
@@ -0,0 +1,363 @@
+		
+		my $log = $self->{Spec}->{Log};
+		
+		my $default = "0"; #setup the default msg flag
+		if ($dir) {
+			if (! chdir $dir) {
+				$self->{Spec}->{amin_error} = "red";
+				my $text = "Unable to change directory to $dir. Reason: $!";
+				$default = 1;
+				$log->error_message($text);
+			}
+		}
+		
+		my $state = 0;
+		foreach my $ip (@$xflag){
+			if (!$ip) {next;};
+			if (($ip =~ /^-/) || ($ip =~ /^--/)) {
+				push @flag, $ip;
+			} else {	
+				if ($state == 0) {
+					if ($ip eq "help") {
+						$flag = "--" . $ip;
+					} elsif ($ip eq "version") {
+						$flag = "--" . $ip;
+					} else {
+						$flag = "-" . $ip;
+					}
+					$state = 1;
+				} else {
+					if ($ip eq "help") {
+						$flag = " --" . $ip;
+					} elsif ($ip eq "version") {
+						$flag = " --" . $ip;
+					} else {
+						$flag = " -" . $ip;
+					}
+				}
+				push @flag, $flag;
+			}
+		}
+		
+		if ($block_size) {
+			$flag = "-B " . $block_size;
+			push @flag, $flag;
+		}
+	
+		if ($type) {
+			$flag = "-t " . $type;
+			push @flag, $flag;
+		}
+		if ($exclude_from) {
+			$flag = "-X " . $exclude_from;
+			push @flag, $flag;
+		}
+		foreach my $ip (@$xparam){
+			push @param, $ip;
+		}
+		$acmd{'CMD'} = "du";
+		$acmd{'FLAG'} = \@flag;
+		$acmd{'PARAM'} = \@param;
+		if ($self->{'ENV_VARS'}) {
+			$acmd{'ENV_VARS'} = $self->{'ENV_VARS'};
+		}
+
+		my $cmd = $self->amin_command(\%acmd);
+
+		if ($cmd->{TYPE} eq "error") {
+			$self->{Spec}->{amin_error} = "red";
+			my $text = "DU Failed. Reason: $cmd->{ERR}";
+			$self->text($text);
+
+			$default = 1;
+			$log->error_message($text);
+			if ($cmd->{ERR}) {
+				$log->ERR_message($cmd->{ERR});
+			}
+		}
+		if (($cmd->{TYPE} eq "out") || ($cmd->{TYPE} eq "both")) {
+			my $otext = "DU was successful.";
+			my $etext = " There was also some error text $cmd->{ERR}";
+			$etext = $otext . $etext; 
+			$default = 1;
+			if ($cmd->{TYPE} eq "out") {
+				$log->success_message($otext);
+				$log->OUT_message($cmd->{OUT});
+			} else {
+				$log->success_message($etext);
+				$log->OUT_message($cmd->{OUT});
+				$log->ERR_message($cmd->{ERR});
+				
+			}
+		}
+		if ($default == 0) {
+			my $text = "there was no messages?";
+			$log->error_message($text);
+		}
+		#reset this command
+		$self->{DIR} = undef;
+		$self->{BLOCK_SIZE} = undef;
+		$self->{TYPE} = undef;
+		$self->{EXCLUDE_FROM} = undef;
+		$self->{FLAG} = [];
+		$self->{PARAM} = [];
+		$self->{COMMAND} = undef;
+		$self->{ATTRS} = undef;
+		$self->{ENV_VARS} = [];
+		$self->{ELEMENT} = undef;
+		$self->SUPER::end_element($element);
+	} else {
+		$self->SUPER::end_element($element);
+	}
+}
+
+
+sub block_size {
+	my $self = shift;
+	$self->{BLOCK_SIZE} = shift if @_;
+	return $self->{BLOCK_SIZE};
+}
+
+sub type {
+	my $self = shift;
+	$self->{TYPE} = shift if @_;
+	return $self->{TYPE};
+}
+
+sub exclude_from {
+	my $self = shift;
+	$self->{EXCLUDE_FROM} = shift if @_;
+	return $self->{EXCLUDE_FROM};
+}
+
+sub filter_map {
```

Comment:
Whats this needed for in the command filters @bdumm2 ?

Reply:
filter_map is being reworked. at moment it is staying same as is...